ml-agents/setup.py

```diff
@@ -66,7 +66,8 @@ def run(self):
         "pyyaml>=3.1.0",
         # Windows ver. of PyTorch doesn't work from PyPi. Installation:
         # https://github.com/Unity-Technologies/ml-agents/blob/main/docs/Installation.md#windows-installing-pytorch
-        'torch>=1.6.0,<1.8.0;platform_system!="Windows"',
+        "torch>=1.8.0,<1.9.0;(platform_system!='Windows' and python_version>='3.9')",
```

Comment:
Maybe add a comment about torch only working on py3.9 for 1.8.0 and above (and link to the issue)